c/src/ml-api-inference-pipeline.c

```diff
@@ -2809,8 +2809,10 @@ ml_pipeline_custom_invoke (void *data, const GstTensorFilterProperties * prop,
 done:
   g_mutex_unlock (&c->lock);
   /* NOTE: DO NOT free tensor data */
-  g_free (in_data);
-  g_free (out_data);
+  if (in_data)
+    _ml_tensors_data_destroy_internal (in_data, FALSE);
+  if (out_data)
+    _ml_tensors_data_destroy_internal (out_data, FALSE);
```

Comment:
You don't need null-check for in_data and out_data
```_ml_tensors_data_create_no_alloc``` won't return OK if it is NULL.